src/actions/filterActions.js

```diff
@@ -0,0 +1,122 @@
+import queryString from 'query-string'
+import deepEquals from 'deep-equal'
+
+import * as eventActions from './eventActions'
+
+export const CHANGE_EVENT_FILTER = 'CHANGE_EVENT_FILTER'
+
+/*SHAPE OF FILTERS
+filters = {
+    incidentId: 0,
+    eventTypes: [{id: 0, name: testName}, ...]
+    occurredStart: someDateTime,
+    occurredEnd: someDateTime
+}*/
+
+export const changeEventFilter = (history) => (filter) => {
+    getUrlFromFilter(history, filter)
+    return {
+        type: CHANGE_EVENT_FILTER,
+        filter
+    }
+}
+
+export const addFilter = (history) => (filter, eventType) => (dispatch) => {
```

Comment:
Since we're not dispatching multiple actions from this, it can lose the extra dispatch call and instead of passing the result of applyFilter to dispatch, just return applyFilter(history)(oldFilter, newFilter)

Reply:
Fixed